fancy_tests/tests/test_views.py

```diff
@@ -298,3 +298,22 @@ def test_cache_backends(self):
             "Random:(\w+)", response.content.decode("utf8")
         )[0]
         ok_(random_string_1 != random_string_2)
+
+    def test_cache_dummy_backend(self):
+        request = self.factory.get("/anything")
+
+        response = views.home8(request)
+        eq_(response.status_code, 200)
+        ok_(re.findall("Random:\w+", response.content.decode("utf8")))
+        random_string_1 = re.findall(
+            "Random:(\w+)", response.content.decode("utf8")
+        )[0]
+
+        # clear second cache backend
+        caches["dummy_backend"].clear()
```

Comment:
And if deleted, the test on line 319 would still be true.

Reply:
Oh I see. The tests fail for DummyCache under Python 3.10. Weird.

Reply:
Found the issue - thank you for taking a look. My updates to the middleware were from the `master` branch of Django which included this commit:

https://github.com/django/django/commit/3ff7b15bb79f2ee5b7af245c55ae14546243bb77

The commit removes `self.cache = caches[self.cach_alias]` from all three middleware classes and replaces them with an `@property` decorator on only two middleware classes - omitting decorator on the base middleware class. However, Django 4.0 doesn't include those `@property` decorators so the line in my latest commit is required.

This will likely be something we'll need to address when the next Django release includes the `@property` decorators via the commit above.